model/src/pyrenew/deterministic/process.py

```diff
@@ -29,14 +30,26 @@ def sample(
         Returns
         -------
         tuple
-            Containing the stored values during construction.
+            Containing the stored values during construction wrapped in a SampledValue.
         """
 
         res, *_ = super().sample(**kwargs)
 
-        dif = duration - res.shape[0]
+        dif = duration - res.value.shape[0]
 
         if dif > 0:
-            return (jnp.hstack([res, jnp.repeat(res[-1], dif)]),)
-
-        return (res[:duration],)
+            return (
+                SampledValue(
+                    jnp.hstack([res.value, jnp.repeat(res.value[-1], dif)]),
+                    t_start=self.t_start,
+                    t_unit=self.t_unit,
+                ),
+            )
+
+        return (
+            SampledValue(
+                value=res.value[:duration],
+                t_start=self.t_start,
+                t_unit=self.t_unit,
+            ),
+        )
```

Comment:
Double return here. Typo?

Reply:
I don't think so. Both return are different.

Reply:
Oh I see one is inside of an `if`. For clarity, I'd suggest this pattern

```python
if <condition>:
      result = <value>
else:
     result = <other value>
return result
```